cache_diff_derive/src/cache_diff_field.rs

```diff
@@ -0,0 +1,387 @@
+//! Represents a field on a struct i.e. `version: String` for implementing the CacheDiff trait
+//!
+//! A `syn::Field`` is can be a named or un-named value (in the case of a tuple struct) or
+//! also represent an Enum. In the case of CacheDiff, we only deal in named structs.
+//!
+//! A field can have zero or more `#[cache_diff()]` attribute annotations for example:
+//!
+//! ```text
+//! #[cache_diff(rename = "Ruby Version")]
+//! name: String
+//! ```
+//!
+//! These attributes look similar to "container" attributes but are different:
+//!
+//! ```text
+//! #[cache_diff(custom = custom_diff)]
+//! struct Metadata {
+//!     // ...
+//! }
+//! ```
+//!
+//! Container attributes are handled by [CacheDiffContainer] and associated files.
+//!
+//! Each comma separated attribute is parsed into a [ParsedAttribute] enum and that information is
+//! combined to form a full [ParsedField].
+//!
+//! A one or more [ParsedField::Active]-s lives inside of a [CacheDiffContainer].
+
+use std::str::FromStr;
+use strum::IntoEnumIterator;
+use syn::{punctuated::Punctuated, spanned::Spanned, Field, Ident, PathArguments, Token};
+
+#[derive(Debug, PartialEq)]
+pub(crate) enum ParsedField {
+    IgnoredCustom,
+    IgnoredOther,
+    Active(ActiveField),
+}
+
+#[derive(Debug, PartialEq)]
+pub(crate) struct ActiveField {
+    /// What the user will see when this field differs and invalidates the cache
+    /// i.e. `age: usize` will be `"age"``
+    pub(crate) name: String,
+    /// The function to use when rendering values on the field
+    /// i.e. `age: 42` will be `"42"`
+    pub(crate) display_fn: syn::Path,
+    /// The proc-macro identifier for a field i.e. `name: String` would be a programatic
+    /// reference to `name` that can be used along with `quote!` to produce code
+    pub(crate) field_identifier: Ident,
+}
+
+impl ParsedField {
+    pub(crate) fn from_field(field: &Field) -> syn::Result<Self> {
```

Comment:
nice, i like what you're doing here but you could save some effort by using [darling](https://github.com/TedDriggs/darling) to handle that conversion of AST -> struct.